packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -111,8 +114,13 @@ export class GarbageCollector implements IGarbageCollector {
 	private readonly initializeGCStateFromBaseSnapshotP: Promise<void>;
 	// The GC details generated from the base snapshot.
 	private readonly baseGCDetailsP: Promise<IGarbageCollectionDetailsBase>;
+
 	// Map of node ids to their unreferenced state tracker.
-	private readonly unreferencedNodesState: Map<string, UnreferencedStateTracker> = new Map();
+	private readonly unreferencedNodesState: UnreferencedStateTrackerMap =
+		new UnreferencedStateTrackerMap();
+	/** Nodes who were unreferenced at the last GC run. Needed to generate GC stats (nodes updated) */
+	private unreferencedNodesAtLastRun: Set<string> = new Set();
```

Comment:
It's unfortunate that we have to track this for just generating stats. However, I am hoping this is temporary because the only reason to have this stat is to validate that we do incremental summary for unchanged data stores only. Today, the GC state change is a reason to re-summarize and hence we need this info. We may not need to do that soon and this should go away then.

Reply:
Good suggestion to keep using the keys in the Map for this, and just hold off on deleting (just stop tracking).